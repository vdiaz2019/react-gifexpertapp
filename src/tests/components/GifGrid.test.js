import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid />>', () => {
    const category = 'Samurai X';

    test('Debe mostrarse correctamente el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = { category } />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargar imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: '123',
            title: 'cualquierCosa.jpg',
            url: 'https://localhost.com'
        },
        {
            id: '456',
            title: 'cualquierCosa.jpg',
            url: 'https://localhost.com'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category = { category } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
