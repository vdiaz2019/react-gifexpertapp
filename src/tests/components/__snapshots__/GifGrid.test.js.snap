// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid />> Debe de mostrar items cuando se cargar im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Samurai X
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="cualquierCosa.jpg"
      url="https://localhost.com"
    />
    <GifGridItem
      id="456"
      key="456"
      title="cualquierCosa.jpg"
      url="https://localhost.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas del componente <GifGrid />> Debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Samurai X
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
